eclipse-collections-api/src/main/java/org/eclipse/collections/api/RichIterable.java

```diff
@@ -194,6 +195,50 @@ default T getOnly()
         return this.anySatisfy(each -> Objects.equals(value, function.valueOf(each)));
     }
 
+    /**
+     * Returns true if any of the elements in source is contained in this collection.
+     *
+     * @since 11.0
+     */
+    default boolean containsAny(Collection<?> source)
+    {
+        return source instanceof RichIterable ? ((RichIterable<?>)source).anySatisfy(this::contains) :
+                source.stream().anyMatch(this::contains);
+    }
+
+    /**
+     * Returns true if none of the elements in source are contained in this collection.
+     *
+     * @since 11.0
+     */
+    default boolean containsNone(Collection<?> source)
+    {
+        return source instanceof RichIterable ? ((RichIterable<?>)source).noneSatisfy(this::contains) :
+                source.stream().noneMatch(this::contains);
+    }
+
+    /**
+     * Returns true if any of the elements in source is contained in this collection.
+     *
+     * @since 11.0
+     */
+    default boolean containsAnyIterable(Iterable<?> source)
+    {
+        return source instanceof RichIterable ? ((RichIterable<?>)source).anySatisfy(this::contains) :
+                StreamSupport.stream(source.spliterator(), false).anyMatch(this::contains);
+    }
+
+    /**
+     * Returns true if none of the elements in source are contained in this collection.
+     *
+     * @since 11.0
+     */
+    default boolean containsNoneIterable(Iterable<?> source)
+    {
+        return source instanceof RichIterable ? ((RichIterable<?>)source).noneSatisfy(this::contains) :
+                StreamSupport.stream(source.spliterator(), false).noneMatch(this::contains);
+    }
+
     /**
      * Returns true if all elements in source are contained in this collection.
      *
```

Comment:
If these methods have default implementations, then containsAll() and its overloads probably ought to as well.

Reply:
We can do followups like this as a separate PR, as it is a lot more work most likely.

Reply:
It's just the next three methods on the next few lines.